### PR TITLE
crd gen: hide from docs regardless of where in the comment it is

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Review by ericvn — APPROVED** [review]
Will likely want this change and #2699 within the same build-tools in the chance there was a change added by #2699 that was supposed to be hidden.

File: cmd/cue-gen/genoapi.go [hunk 1]
```diff
@@ -426,7 +426,7 @@ func (x *builder) genOpenAPI(name string, inst cue.Value) (cue.Value, error) {
 			if doc.Text() == "" {
 				continue
 			}
-			if strings.HasPrefix(doc.Text(), "$hide_from_docs") {
+			if strings.Contains(doc.Text(), "$hide_from_docs") {
```
Comment (jwendell) [inline cmd/cue-gen/genoapi.go:429]: what if this line is commented? Is that possible?

Comment (howardjohn) [reply cmd/cue-gen/genoapi.go:429]: Not really, the line is already a (protobuf) comment like `// $hide_from_docs`. So there isn't really a mechanism to have a commented-comment

Comment (jwendell) [reply cmd/cue-gen/genoapi.go:429]: ah ok :)

**Pull request merged**